src/Trace/Trace.php

```diff
@@ -0,0 +1,211 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Core\ValidateTrait;
+use Google\Cloud\Trace\Connection\ConnectionInterface;
+
+/**
+ * This plain PHP class represents a Trace resource. Traces belong to a
+ * project and have many TraceSpans. See:
+ * https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#resource-trace
+ */
+class Trace
+{
+    use ValidateTrait;
+
+    /**
+     * @var ConnectionInterface Represents a connection to Stackdriver Trace.
+     */
+    private $connection;
+
+    /**
+     * The id of the project this trace belongs to.
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * The trace id for this trace. 128-bit numeric formatted as a 32-byte hex string
+     * @var string
+     */
+    private $traceId;
+
+    /**
+     * List of TraceSpans to report
+     * @var TraceSpan[]
+     */
+    private $spans = [];
+
+    /**
+     * Instantiate a new Trace instance.
+     *
+     * @param ConnectionInterface $connection The connection to Stackdriver Trace.
+     * @param string $projectId The id of the project this trace belongs to.
+     * @param string $traceId [optional] The id of the trace. If not provided, one will be generated
+     *        automatically for you.
+     * @param array $spans [optional] Array of TraceSpan constructor arguments. See
+     *        {@see Google\Cloud\Trace\TraceSpan::__construct()} for configuration details.
+     * }
+     */
+    public function __construct(ConnectionInterface $connection, $projectId, $traceId = null, $spans = null)
+    {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        $this->traceId = $traceId ?: $this->generateTraceId();
+        if ($spans) {
+            $this->spans = array_map(function ($span) {
+                return new TraceSpan($span);
+            }, $spans);
+        }
+    }
+
+    /**
+     * Retrieves the trace's id.
+     *
+     * @return string
+     */
+    public function traceId()
+    {
+        return $this->traceId;
+    }
+
+    /**
+     * Retrieves the trace's projectId.
+     *
+     * @return string
+     */
+    public function projectId()
```

Comment:
Is it something we can hide from users? Is there a case where the projectId would vary from the one set when instantiating the client?